sleap/nn/inference.py

```diff
@@ -2198,6 +2199,322 @@ def call(
         return crop_output
 
 
+@attr.s(auto_attribs=True)
+class CentroidPredictor(Predictor):
+    """Centroid multi-instance predictor.
+
+    This high-level class handles initialization, preprocessing and tracking using a
+    trained top-down multi-instance SLEAP model.
+
+    This should be initialized using the `from_trained_models()` constructor or the
+    high-level API (`sleap.load_model`).
+
+    Attributes:
+        centroid_config: The `sleap.nn.config.TrainingJobConfig` containing the metadata
+            for the trained centroid model.
+        centroid_model: A `sleap.nn.model.Model` instance created from the trained
+            centroid model. If `None`, ground truth centroids will be used if available
+            from the data source.
+        inference_model: A `sleap.nn.inference.CentroidInferenceModel` that wraps a
+            trained `tf.keras.Model` to implement preprocessing, centroid detection, and
+            cropping.
+        pipeline: A `sleap.nn.data.Pipeline` that loads the data and batches input data.
+            This will be updated dynamically if new data sources are used.
+        tracker: A `sleap.nn.tracking.Tracker` that will be called to associate
+            detections over time. Predicted instances will not be assigned to tracks if
+            if this is `None`.
+        batch_size: The default batch size to use when loading data for inference.
+            Higher values increase inference speed at the cost of higher memory usage.
+        peak_threshold: Minimum confidence map value to consider a local peak as valid.
+        integral_refinement: If `True`, peaks will be refined with integral regression.
+            If `False`, `"local"`, peaks will be refined with quarter pixel local
+            gradient offset. This has no effect if the model has an offset regression
+            head.
+        integral_patch_size: Size of patches to crop around each rough peak for integral
+            refinement as an integer scalar.
+        max_instances: If not `None`, discard instances beyond this count when
+            predicting, regardless of whether filtering is done at the tracking stage.
+            This is useful for preventing extraneous instances from being created when
+            tracking is not being applied.
+    """
+
+    centroid_config: TrainingJobConfig
+    centroid_model: Optional[Model] = attr.ib(default=None)
```

Comment:
We need to validate that the model either:
1. is a single node skeleton
2. specifies an anchor point